File: work/nev-differential/variance-20260924/run-jev.py
#!/usr/bin/env python3
"""Third Jev run for bead jev-rf57: the 662 jev-sec-bench injection rows, jev-1.13.0.

Questions, state and corpus come from work/nev-differential/run_diff.py, imported and not edited:
build_questions() (the upstream InjectionBattery Noul + Score), ASSISTANT, and SRC
(jev-sec-bench/results/injection.json). The client is the official typesafe_sdk AsyncTypeSafeClient
pinned to jev-1.13.0 with RetryPolicy(), concurrency 8. Rows go to rows-jev-run3.jsonl beside this
file, one per id, in the fresh-20260923 row shape (id, label, p, sev, sev_conf, lat_ms, in_tok,
out_tok, model). A rerun resumes rows that have no answer. Never prints a key.
Bar: docs/demos/upstream-repro/injection-variance-20260924.md (committed before any call).

Run:
  infisical run --silent --projectId=42b194c3-89d7-4ebb-895f-dd77ddf005ba -- \
    upstream/typesafe-ai/system-one-adapter-python/.venv/bin/python \
    work/nev-differential/variance-20260924/run-jev.py
"""

import asyncio
import importlib.util
import json
import os
import sys
import time

HERE = os.path.dirname(os.path.abspath(__file__))
_spec = importlib.util.spec_from_file_location(
    "run_diff", os.path.join(os.path.dirname(HERE), "run_diff.py")
)
RD = importlib.util.module_from_spec(_spec)
_spec.loader.exec_module(RD)

from typesafe_sdk import AsyncTypeSafeClient, RetryPolicy  # noqa: E402

MODEL = "jev-1.13.0"
OUT = os.path.join(HERE, "rows-jev-run3.jsonl")


def done():
    ids = set()
    if os.path.exists(OUT):
        for line in open(OUT):
            if line.strip():
                r = json.loads(line)
                if "p" in r:
                    ids.add(r["id"])
    return ids


async def main(concurrency=8, bar_path=None, repo=None):
    root = os.path.dirname(os.path.dirname(os.path.dirname(HERE)))
    sys.path.insert(0, os.path.join(root, "work/sr-adopt"))
    from phase_gate import call_after_bar

    bar = bar_path or os.path.join(
        root, "docs/demos/upstream-repro/injection-variance-20260924.md"
    )
    call_after_bar(bar, lambda: None, repo=repo or root)
    if not os.environ.get("TYPESAFE_API_KEY"):
        print(
            "unconfigured: TYPESAFE_API_KEY is not set, no call made", file=sys.stderr
        )
        return 2
    samples = json.load(open(RD.SRC))["samples"]
    rows = [(f"inj-{i:04d}", s["text"], int(s["label"])) for i, s in enumerate(samples)]
    assert len(rows) == 662, f"corpus drift: {len(rows)}"
    have = done()
    todo = [r for r in rows if r[0] not in have]
    print(f"jev run3: {len(todo)} to run, {len(have)} resumed", file=sys.stderr)
    questions = RD.build_questions()
    sem = asyncio.Semaphore(concurrency)
    ok = failed = 0
    async with AsyncTypeSafeClient(model=MODEL, retry=RetryPolicy()) as client:

        async def one(rid, text, label):
            async with sem:
                t0 = time.time()
                rec = {"id": rid, "label": label}
                try:
                    resp = await asyncio.wait_for(
                        client.system_one(
                            {"assistant": RD.ASSISTANT, "user_message": text}, questions
                        ),
                        timeout=90,
                    )
                    sev = resp.scores["severity"]
                    rec.update(
                        p=float(resp.nouls["injection"].noul),
                        sev=float(sev.score),
                        sev_conf=float(sev.confidence),
                        lat_ms=int((time.time() - t0) * 1000),
                        in_tok=resp.usage.input_tokens,
                        out_tok=resp.usage.output_tokens,
                        model=resp.model,
                    )
                except Exception as e:  # noqa: BLE001 - recorded, never scored as an answer
                    rec["error"] = f"{type(e).__name__}: {str(e)[:200]}"
                return rec

        for coro in asyncio.as_completed([one(*r) for r in todo]):
            rec = await coro
            with open(OUT, "a") as f:
                f.write(json.dumps(rec, ensure_ascii=False) + "\n")
            if "error" in rec:
                failed += 1
            else:
                ok += 1
    print(f"done: ok={ok} failed={failed}", file=sys.stderr)
    return 3 if failed else 0


if __name__ == "__main__":
    sys.exit(asyncio.run(main()))

File: work/jev-toolout-flag/run-jev.mjs
// Jev cells for bead jev-qip: the jev_screen QUESTION with the `assistant` description withheld
// (state {user_message}), pinned jev-1.13.0, cut 0.5.
// Bar: docs/demos/upstream-repro/jev-toolout-flag-20260924.md (committed before any call).
//
//   node run-jev.mjs <arm> <corpus>
//   arm    plain     the QUESTION alone (the k9z.5 evidence-withheld state)
//          criteria  the QUESTION plus Noul criteria {true,false} from criteria.json
//   corpus attacks   work/nev-injection/pairs.jsonl (662 public rows, 263 attacks)
//          toolout   work/jev-injection-flag/tool-results-sample.json (300 real tool results)
// plain x toolout is not re-run: the scorer reads work/jev-injection-flag/rows-jev-withheld.jsonl,
// which is exactly that cell (k9z.5, 888efbe).
// Writes rows-jev-<arm>-<corpus>.jsonl next to this file; resumes rows that already have a score;
// a failed row is recorded, never scored.
// Run: infisical run --silent --projectId=42b194c3-89d7-4ebb-895f-dd77ddf005ba -- \
//        node work/jev-toolout-flag/run-jev.mjs criteria attacks
import { appendFileSync, existsSync, readFileSync } from 'node:fs';
import { askJev } from '../jev-client/src/index.ts';
import { QUESTION } from '../jev-injection-flag/seat-question.mjs';
import { loadCorpus } from './corpus.mjs';

const HERE = new URL('.', import.meta.url);
const [ARM, CORPUS] = process.argv.slice(2);
if (!['plain', 'criteria'].includes(ARM) || !['attacks', 'toolout'].includes(CORPUS)) {
  console.error('usage: run-jev.mjs plain|criteria attacks|toolout');
  process.exit(64);
}
if (ARM === 'plain' && CORPUS === 'toolout') {
  console.error('plain x toolout is committed as work/jev-injection-flag/rows-jev-withheld.jsonl; not re-run');
  process.exit(64);
}
const OUT = new URL(`rows-jev-${ARM}-${CORPUS}.jsonl`, HERE);
const CUT = 0.5;
const MODEL = 'jev-1.13.0';
const CONCURRENCY = Number(process.env.CONCURRENCY ?? 8);
const crit = JSON.parse(readFileSync(new URL('criteria.json', HERE), 'utf8'));
const question = ARM === 'plain'
  ? QUESTION
  : { instructions: QUESTION, criteria: { true: crit.true, false: crit.false } };

if (!process.env.TYPESAFE_API_KEY) {
  console.error('unconfigured: TYPESAFE_API_KEY is not set — no network call made');
  process.exit(2);
}

const rows = loadCorpus(CORPUS);
const done = new Set();
if (existsSync(OUT)) {
  for (const line of readFileSync(OUT, 'utf8').split('\n')) {
    if (!line.trim()) continue;
    const r = JSON.parse(line);
    if (typeof r.p === 'number') done.add(r.i);
  }
}
const todo = rows.filter((r) => !done.has(r.i));
console.error(`jev ${ARM}/${CORPUS}: ${todo.length} to run, ${done.size} resumed, concurrency ${CONCURRENCY}`);

let next = 0, ok = 0, failed = 0;
async function worker() {
  while (next < todo.length) {
    const row = todo[next++];
    const r = await askJev({ state: { user_message: row.text }, questions: { inj: question }, model: MODEL, timeoutMs: 20000 });
    const out = { i: row.i, id: row.id, label: row.label, arm: ARM, model: r.model, latencyMs: r.latencyMs };
    if (r.ok && typeof r.scores.inj === 'number') {
      Object.assign(out, { p: r.scores.inj, flag: r.scores.inj >= CUT, usage: r.usage ?? null });
      ok++;
    } else {
      Object.assign(out, { error: r.ok ? 'incomplete-scores' : `${r.reason}: ${r.error}` });
      failed++;
    }
    appendFileSync(OUT, JSON.stringify(out) + '\n');
    if ((ok + failed) % 50 === 0) console.error(`  ${ok + failed}/${todo.length} ok=${ok} failed=${failed}`);
  }
}
await Promise.all(Array.from({ length: CONCURRENCY }, worker));
console.error(`done: ok=${ok} failed=${failed}`);
process.exit(failed > 6 ? 3 : 0);
